knowledgegpt/extractors/helpers.py

```diff
@@ -9,3 +9,32 @@ def check_embedding_extractor(embedding_extractor, embedding_extractor_acceptabl
 
         raise Exception(f"Embedding Extractor is not allowed. "
                         f"Please choose one of : {embedding_extractor_acceptable_list}")
+
+
+def check_model_lang(model_lang, model_lang_acceptable_list=None):
+    if model_lang_acceptable_list is None:
+        model_lang_acceptable_list = ["en", "tr"]
+
+    if not isinstance(model_lang, str):
+        raise Exception("Model Lang must be a string")
+
+    if model_lang not in model_lang_acceptable_list:
+        raise Exception(f"Model Lang is not allowed. "
+                        f"Please choose one of : {model_lang_acceptable_list}")
+    
+
+def check_index_type(index_type, index_type_acceptable_list=None):
+    if index_type_acceptable_list is None:
+        index_type_acceptable_list = ["basic", "faiss"]
+
+    if not isinstance(index_type, str):
+        raise Exception("Index Type must be a string")
+
+    if index_type not in index_type_acceptable_list:
+        raise Exception(f"Index Type is not allowed. "
+                        f"Please choose one of : {index_type_acceptable_list}")
+    
+def check_save_index(save_index, index_path):
+    if (save_index is True and index_path is None):
+        raise Exception("Index Path must be provided when save_index is True")
```

Comment:
if index path none then save index means false, if index path true then we do not need to check save_index parameter